### PR TITLE
Implement list tracking

### PR DESCRIPTION
**Pull request by anhmht**
#70  
- Create List of tracking tracking/components/ListTrackingResult.jsx
screencapture-localhost-3000-trackings-2019-03-23-19_21_54

 - Create TrackingFilterForm.jsx
 - Change status of Tracking to OnHold and Returning with drop down button

### CODE REVIEW

File: imports/ui/modules/tracking/components/TrackingFilterForm.jsx [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+    flights.forEach((flight) => {
+      data.push(flight.flightId);
+    });
+
+    return (
+      <Form onSubmit={handleSubmit(this.onSubmit)}>
+        <Row gutter={16}>
+          <Col span={8}>
+            <InputField
+              name="flightId"
```
Comment (hoadinh2010) [inline imports/ui/modules/tracking/components/TrackingFilterForm.jsx]: You have written a custom auto complete component input, Why you still using this kind of input type ?

Comment (hoadinh2010) [reply imports/ui/modules/tracking/components/TrackingFilterForm.jsx]: But, I still don't know why you are using AutoComplete component. You must use Select component of ant design. Using AutoComplete make you cannot choose another flight when already chosen one.

File: imports/ui/modules/tracking/components/TrackingFilterForm.jsx [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+import { Meteor } from 'meteor/meteor';
+import { reduxForm } from 'redux-form';
+import { REDUX_FORM_NAMES } from '../../../../../lib/enums/general';
+import { FORM_TYPES } from '../../../../../lib/enums/form';
+import Flights from '../../../../../lib/collections/flights';
+import InputField from '../../common/form/InputField';
+import { TRACKING_STATUSES, TRACKING_STATUSES_TEXT } from '../../../../../lib/enums/tracking';
+import { ROLES } from '../../../../../lib/enums/roles';
+
+const trackingStatus = [
```
Comment (hoadinh2010) [inline imports/ui/modules/tracking/components/TrackingFilterForm.jsx]: Using Select component and don't use map data like that, just loop on the enum entity to build the option

File: imports/ui/modules/tracking/components/TrackingFilterForm.jsx [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+    filter: shape(),
+  }
+
+  static defaultProps = {
+    filter: null,
+  }
+
+  state = {}
+
+  mapStatusValue = (value) => {
```
Comment (hoadinh2010) [inline imports/ui/modules/tracking/components/TrackingFilterForm.jsx]: Above issue, this funtcion doesn't make sense with me

File: imports/api/publications/tracking.js [hunk 4]
```diff
@@ -28,3 +28,13 @@ Meteor.publish('tracking.getListTracking', (filter) => {
     }
   });
 });
+
+Meteor.publish('tracking.getAllTrackings', () => {
```
[the remark this answers is not in the document]
Comment (anhmht) [reply imports/api/publications/tracking.js]: This publication was deleted

File: imports/api/publications/flight.js [hunk 5]
```diff
@@ -18,3 +18,11 @@ Meteor.publish('flight.getFlightDetail', (_id) => {
   return Flights.find({ _id }, {
   });
 });
+
+Meteor.publish('flight.getFlightDetailByFlightId', (flightId) => {
```
[the remark this answers is not in the document]
Comment (anhmht) [reply imports/api/publications/flight.js]: This publication was deleted, At the begining I wrote this fuction but never use, so this was deleted

File: imports/ui/modules/tracking/components/ListTrackingPage.jsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+const mapStateToProps = (state) => {
+  const trackingState = state.tracking;
+  return {
+    filter: trackingState.filter,
+  };
+};
+
+const mapDispatchToProps = dispatch => bindActionCreators(
```
[the remark this answers is not in the document]
Comment (anhmht) [reply imports/ui/modules/tracking/components/ListTrackingPage.jsx]: This unused function was deleted

File: imports/ui/modules/tracking/components/ListTrackingResult.jsx [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+
+      </div>
+    );
+  }
+}
+const mapStateToProps = state => ({
+  valueForms: getFormValues(REDUX_FORM_NAMES.TRACKING_FILTER_FORM)(state),
+});
+
+const mapDispatchToProps = dispatch => bindActionCreators(
```
[the remark this answers is not in the document]
Comment (anhmht) [reply imports/ui/modules/tracking/components/ListTrackingResult.jsx]: Same as above, unused function was deleted

File: imports/ui/modules/tracking/components/ListTrackingResult.jsx [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+export default compose(
+  reduxForm({
+    form: REDUX_FORM_NAMES.TRACKING_FILTER_FORM,
+  }),
+  connect(mapStateToProps, mapDispatchToProps),
+  withTracker((props) => {
+    let trackings = null;
+    if (props.filter) {
+      const { filter } = props;
+      const handleFlight = Meteor.subscribe('flight.getFlightList');
```
[the remark this answers is not in the document]
Comment (anhmht) [reply imports/ui/modules/tracking/components/ListTrackingResult.jsx]: all the publication was moved to `ListTrackingPage`

File: imports/ui/modules/tracking/components/ListTrackingResult.jsx [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+import { Meteor } from 'meteor/meteor';
+import { reduxForm, getFormValues } from 'redux-form';
+import moment from 'moment';
+import { REDUX_FORM_NAMES } from '../../../../../lib/enums/general';
+import { ROLES } from '../../../../../lib/enums/roles';
+import Flights from '../../../../../lib/collections/flights';
+import Trackings from '../../../../../lib/collections/tracking';
+import { TRACKING_STATUSES_TEXT, TRACKING_STATUSES_COLOR, TRACKING_STATUSES } from '../../../../../lib/enums/tracking';
+
+class ListTrackingPage extends React.PureComponent {
```
[the remark this answers is not in the document]
Comment (anhmht) [reply imports/ui/modules/tracking/components/ListTrackingResult.jsx]: The name of file ListTrackingResult was updated

File: imports/ui/modules/tracking/components/ListTrackingResult.jsx [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+    form: REDUX_FORM_NAMES.TRACKING_FILTER_FORM,
+  }),
+  connect(mapStateToProps, mapDispatchToProps),
+  withTracker((props) => {
+    let trackings = null;
+    if (props.filter) {
+      const { filter } = props;
+      const handleFlight = Meteor.subscribe('flight.getFlightList');
+      const handleUser = Meteor.subscribe('user.getCustomersList');
+      const handleTrackings = Meteor.subscribe('tracking.getAllTrackings');
```
[the remark this answers is not in the document]
Comment (anhmht) [reply imports/ui/modules/tracking/components/ListTrackingResult.jsx]: `const handleTrackings = Meteor.subscribe('tracking.getListTracking', props.filter);`
I fixed the subscription

File: imports/ui/modules/tracking/components/TrackingFilterForm.jsx (hunk 1 above)
Comment (anhmht) [reply imports/ui/modules/tracking/components/TrackingFilterForm.jsx]: I changed All the inputField to to select input

File: imports/ui/modules/tracking/components/TrackingFilterForm.jsx (hunk 3 above)
Comment (anhmht) [reply imports/ui/modules/tracking/components/TrackingFilterForm.jsx]: this function was deleted too

File: imports/ui/modules/tracking/components/TrackingFilterForm.jsx [hunk 11]
```diff
@@ -13,29 +13,32 @@ import ActionCreators from '/imports/ui/actions';
...
-  TRACKING_STATUSES_TEXT[TRACKING_STATUSES.IN_TRANSIT],
-  TRACKING_STATUSES_TEXT[TRACKING_STATUSES.US_DELIVERED],
-  TRACKING_STATUSES_TEXT[TRACKING_STATUSES.ON_HOLD],
-  TRACKING_STATUSES_TEXT[TRACKING_STATUSES.PACKED],
-  TRACKING_STATUSES_TEXT[TRACKING_STATUSES.ON_BOARD],
-  TRACKING_STATUSES_TEXT[TRACKING_STATUSES.VN_DELIVERED],
-  TRACKING_STATUSES_TEXT[TRACKING_STATUSES.DOOR_DELIVERED],
-  TRACKING_STATUSES_TEXT[TRACKING_STATUSES.RETURNING],
+  { label: 'Chọn tất cả', value: 'ALL' },
+  { label: TRACKING_STATUSES_TEXT[TRACKING_STATUSES.IN_TRANSIT], value: TRACKING_STATUSES.IN_TRANSIT },
```
Comment (hoadinh2010) [inline imports/ui/modules/tracking/components/TrackingFilterForm.jsx]: Why don't you use loop in this case . ?

Comment (tichondrius) [reply imports/ui/modules/tracking/components/TrackingFilterForm.jsx]: @anhmht , You can use Object.keys to load all keys of Object and .map function

const options = [
  { label: 'Chọn tất cả', value: 'ALL' },
  ...Object.keys(TRACKING_STATUSES)
    .map(trackingStatus => ({
      label: TRACKING_STATUSES_TEXT[trackingStatus],
      value: TRACKING_STATUSES[trackingStatus],
    }))];
}))

Comment (anhmht) [reply imports/ui/modules/tracking/components/TrackingFilterForm.jsx]: thanks for the support

**Pull request merged**